### PR TITLE
Social sqlalchemy

### PR DESCRIPTION
**Pull request by NyuydineBill**
Description:
This pull request adds optional instructions for configuring social_sqlalchemy, a package for SQLAlchemy storage support for Python Social Auth, in Anitya . It provides a code example demonstrating how to set up social_sqlalchemy in a Anitya , including configuring the SQLAlchemy database and initializing social_sqlalchemy. Additionally, it includes a note on starting the development web server with Flask.

Purpose:
The purpose of this pull request is to enhance the contributing guidelines by providing guidance on integrating social_sqlalchemy with Anitya for OAuth and OAuth2 authentication. This addition aims to assist contributors who may want to use social_sqlalchemy in  Anitya project, thereby improving the overall development experience for Anitya contributors.

Changes Made:
Added instructions for configuring social_sqlalchemy in Flask applications to the contributing.rst file.
Included a code example demonstrating the setup process for social_sqlalchemy.
Updated the Python Virtualenv section of the contributing.rst file to include these instructions.
Testing:
These instructions were tested locally by following the provided steps to configure social_sqlalchemy in a Anitya application within a Python virtual environment. The Flask development server was started successfully, indicating that the setup process was correct and functional.

Related Issue:
#1723

### CODE REVIEW

**Comment by Zlopez** [comment]
And also don't forget to add news file for the update. See https://anitya.readthedocs.io/en/latest/contributing.html#release-notes for more info

File: docs/contributing.rst [hunk 1]
```diff
@@ -278,6 +278,22 @@ Create the database, by default it will be a sqlite database located at
 
     $ poetry run python createdb.py
 
+Configure social_sqlalchemy for Anitya if needed. This step is optional and depends on your use case: 
+in config.py::
```
Comment (Zlopez) [inline docs/contributing.rst:282]:
```suggestion
in `anitya/config.py`::
```

File: news/1723.other
Comment (Zlopez) [inline news/1723.other:1]: Ideally this should be a one line file.

**Review by Zlopez — CHANGES_REQUESTED** [review]
Still some changes required, but getting there.

File: news/1723.other
Comment (NyuydineBill) [reply news/1723.other:1]: corrected

**Comment by Zlopez** [comment]
Thanks for fixing the tests! Could you now squash the commits so there is only one commit message?

**Pull request closed without merging**